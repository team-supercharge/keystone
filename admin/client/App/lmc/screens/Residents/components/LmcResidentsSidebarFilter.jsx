import React from 'react'
import PropTypes from 'prop-types'
import { FormInput } from '../../../../elemental'
import Switch from 'react-switch'

const LmcResidentsSidebarFilter = ({ onFormChange, onSwitchChange, isChecked }) => {
    return (
        <div style={styles.filterContainer}>
            <FormInput
                autofocus
                name='nameFilter'
                placeholder="Filter by name"
                style={styles.filter}
                onChange={onFormChange}
            />
            <span style={{ cursor: 'pointer', float: 'right' }} onClick={() => {}}>
                <span 
                    style={{ paddingRight: 8, position: 'relative', top: -5, opacity: 0.6 }}
                    onClick={onSwitchChange}
                >
                    Active residents
                </span>
                    <Switch
                        onChange={onSwitchChange}
                        checked={isChecked}
                        onColor='#e65d78'
                        offColor='#e65d78'
                        handleDiameter={16}
                        uncheckedIcon={false}
                        checkedIcon={false}
                        activeBoxShadow="0px 0px 1px 8px rgba(0, 0, 0, 0.1)"
                        height={20}
                        width={40}
                    />
                <span 
                    style={{ paddingLeft: 8, position: 'relative', top: -5, opacity: 0.6 }}
                    onClick={onSwitchChange}
                >
                    All
                </span>
            </span>
        </div>
    )
}

const styles = {
    filter: {
        borderRadius: '3px',
        height: '40px',
        marginBottom: '15px',
    },
    filterContainer: {
        padding: '10px',
        display: 'block',
        backgroundColor: 'white',
    },
}

LmcResidentsSidebarFilter.propTypes = {
    onFormChange: PropTypes.func.isRequired,
    onSwitchChange: PropTypes.func.isRequired,
    isChecked: PropTypes.bool.isRequired,
}

export default LmcResidentsSidebarFilter