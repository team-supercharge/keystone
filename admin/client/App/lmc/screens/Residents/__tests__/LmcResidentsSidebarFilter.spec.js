import React from 'react'
import { shallow } from 'enzyme'
import LmcResidentsSidebarFilter from '../components/LmcResidentsSidebarFilter.jsx'

describe('LmcResidentsSidebarFilter', () => {
    let wrapper
    let onFormMock = jest.fn()
    let onSwitchMock = jest.fn()

    beforeEach(() => {
        wrapper = shallow(
            <LmcResidentsSidebarFilter
                onFormChange={onFormMock}
                onSwitchChange={onSwitchMock}
                isChecked={true}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('triggers its form change function', () => {
        wrapper.find('FormInput').simulate('change')
        expect(onFormMock).toBeCalledTimes(1)
    })

    it('triggers its switch change function', () => {
        wrapper.find('ReactSwitch').simulate('change')
        expect(onSwitchMock).toBeCalledTimes(1)
    })
})