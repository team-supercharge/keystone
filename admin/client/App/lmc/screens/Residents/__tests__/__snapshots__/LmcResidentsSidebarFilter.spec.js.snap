// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcResidentsSidebarFilter renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
      Object {
        "backgroundColor": "white",
        "display": "block",
        "padding": "10px",
      }
    }
  >
    <FormInput
      autofocus={true}
      name="nameFilter"
      onChange={[MockFunction]}
      placeholder="Filter by name"
      size="default"
      style={
        Object {
          "borderRadius": "3px",
          "height": "40px",
          "marginBottom": "15px",
        }
      }
      type="text"
    />
    <span
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "float": "right",
        }
      }
    >
      <span
        onClick={[MockFunction]}
        style={
          Object {
            "opacity": 0.6,
            "paddingRight": 8,
            "position": "relative",
            "top": -5,
          }
        }
      >
        Active residents
      </span>
      <ReactSwitch
        activeBoxShadow="0px 0px 1px 8px rgba(0, 0, 0, 0.1)"
        boxShadow={null}
        checked={true}
        checkedIcon={false}
        disabled={false}
        handleDiameter={16}
        height={20}
        offColor="#e65d78"
        offHandleColor="#fff"
        onChange={[MockFunction]}
        onColor="#e65d78"
        onHandleColor="#fff"
        uncheckedIcon={false}
        width={40}
      />
      <span
        onClick={[MockFunction]}
        style={
          Object {
            "opacity": 0.6,
            "paddingLeft": 8,
            "position": "relative",
            "top": -5,
          }
        }
      >
        All
      </span>
    </span>
  </div>,
  "nodes": Array [
    <div
      style={
        Object {
          "backgroundColor": "white",
          "display": "block",
          "padding": "10px",
        }
      }
    >
      <FormInput
        autofocus={true}
        name="nameFilter"
        onChange={[MockFunction]}
        placeholder="Filter by name"
        size="default"
        style={
          Object {
            "borderRadius": "3px",
            "height": "40px",
            "marginBottom": "15px",
          }
        }
        type="text"
      />
      <span
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "float": "right",
          }
        }
      >
        <span
          onClick={[MockFunction]}
          style={
            Object {
              "opacity": 0.6,
              "paddingRight": 8,
              "position": "relative",
              "top": -5,
            }
          }
        >
          Active residents
        </span>
        <ReactSwitch
          activeBoxShadow="0px 0px 1px 8px rgba(0, 0, 0, 0.1)"
          boxShadow={null}
          checked={true}
          checkedIcon={false}
          disabled={false}
          handleDiameter={16}
          height={20}
          offColor="#e65d78"
          offHandleColor="#fff"
          onChange={[MockFunction]}
          onColor="#e65d78"
          onHandleColor="#fff"
          uncheckedIcon={false}
          width={40}
        />
        <span
          onClick={[MockFunction]}
          style={
            Object {
              "opacity": 0.6,
              "paddingLeft": 8,
              "position": "relative",
              "top": -5,
            }
          }
        >
          All
        </span>
      </span>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <LmcResidentsSidebarFilter
        isChecked={true}
        onFormChange={[MockFunction]}
        onSwitchChange={[MockFunction]}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "isChecked": true,
          "onFormChange": [MockFunction],
          "onSwitchChange": [MockFunction],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
            Object {
              "backgroundColor": "white",
              "display": "block",
              "padding": "10px",
            }
          }
        >
          <FormInput
            autofocus={true}
            name="nameFilter"
            onChange={[MockFunction]}
            placeholder="Filter by name"
            size="default"
            style={
              Object {
                "borderRadius": "3px",
                "height": "40px",
                "marginBottom": "15px",
              }
            }
            type="text"
          />
          <span
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "float": "right",
              }
            }
          >
            <span
              onClick={[MockFunction]}
              style={
                Object {
                  "opacity": 0.6,
                  "paddingRight": 8,
                  "position": "relative",
                  "top": -5,
                }
              }
            >
              Active residents
            </span>
            <ReactSwitch
              activeBoxShadow="0px 0px 1px 8px rgba(0, 0, 0, 0.1)"
              boxShadow={null}
              checked={true}
              checkedIcon={false}
              disabled={false}
              handleDiameter={16}
              height={20}
              offColor="#e65d78"
              offHandleColor="#fff"
              onChange={[MockFunction]}
              onColor="#e65d78"
              onHandleColor="#fff"
              uncheckedIcon={false}
              width={40}
            />
            <span
              onClick={[MockFunction]}
              style={
                Object {
                  "opacity": 0.6,
                  "paddingLeft": 8,
                  "position": "relative",
                  "top": -5,
                }
              }
            >
              All
            </span>
          </span>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
            Object {
              "backgroundColor": "white",
              "display": "block",
              "padding": "10px",
            }
          }
        >
          <FormInput
            autofocus={true}
            name="nameFilter"
            onChange={[MockFunction]}
            placeholder="Filter by name"
            size="default"
            style={
              Object {
                "borderRadius": "3px",
                "height": "40px",
                "marginBottom": "15px",
              }
            }
            type="text"
          />
          <span
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "float": "right",
              }
            }
          >
            <span
              onClick={[MockFunction]}
              style={
                Object {
                  "opacity": 0.6,
                  "paddingRight": 8,
                  "position": "relative",
                  "top": -5,
                }
              }
            >
              Active residents
            </span>
            <ReactSwitch
              activeBoxShadow="0px 0px 1px 8px rgba(0, 0, 0, 0.1)"
              boxShadow={null}
              checked={true}
              checkedIcon={false}
              disabled={false}
              handleDiameter={16}
              height={20}
              offColor="#e65d78"
              offHandleColor="#fff"
              onChange={[MockFunction]}
              onColor="#e65d78"
              onHandleColor="#fff"
              uncheckedIcon={false}
              width={40}
            />
            <span
              onClick={[MockFunction]}
              style={
                Object {
                  "opacity": 0.6,
                  "paddingLeft": 8,
                  "position": "relative",
                  "top": -5,
                }
              }
            >
              All
            </span>
          </span>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcResidentsSidebarFilter
    isChecked={true}
    onFormChange={[MockFunction]}
    onSwitchChange={[MockFunction]}
  />,
}
`;
